hub/models.py

```diff
@@ -1597,3 +1599,67 @@ class MapLayer(TypedDict):
 
     def get_layers(self) -> list[MapLayer]:
         return self.layers
+
+
+class MailchimpSource(ExternalDataSource):
+    """
+    A Mailchimp list.
+    """ 
+    api_key = models.CharField(
+        max_length=250,
+        help_text="Mailchimp API key.",
+    )
+    list_id = models.CharField(
+        max_length=250,
+        help_text="The unique identifier for the Mailchimp list.",
+    )
+
+    automated_webhooks = True
+    introspect_fields = True
+
+    class Meta:
+        verbose_name = "Mailchimp list"
+        unique_together = ["list_id", "api_key"]
+
+    @cached_property
+    def client(self) -> MailChimp:
+        # Initializes the MailChimp client 
+        return MailChimp(mc_api=self.api_key)
+    
+        
+    def healthcheck(self):
+        # Checks if the Mailchimp list is accessible
+        list = self.client.lists.get(self.list_id)
+        if list:
+            return True
+        return False
+    
+    def field_definitions(self):
+        merge_fields = self.client.lists.merge_fields.all(list_id=self.list_id, get_all=True)['merge_fields']
+        self.field_definitions = [
+            {
+                'label': field['name'],
+                'value': field['tag'],  # In Mailchimp, the tag is used as a merge variable
+                'description': field.get('help_text', '')
+            }
+            for field in merge_fields
+        ]
+    
+        return self.field_definitions
+
+    
+    def remote_name(self):
+        data = self.client.lists.all(self.list_id, fields="lists.name")
+        list_name = data['lists'][0]['name']
+        return list_name
+   
+    async def fetch_all(self):
+         # Fetches all members in a list and returns their email addresses
+        list = self.client.lists.members.all(self.list_id, fields="members.email_address")
+        return list
```

Comment:
This is almost OK but what's missing, crucially, is the member's `geography_column` which will be important for the data mapping + data importing.

Ideally we'd also get the other merge columns for use in report making, but that's less important.

Reply:
I've updated it to include all merge fields